libs/staging-provider/src/Staging/Workspace/SnowflakeWorkspaceStaging.php

```diff
@@ -8,4 +8,33 @@ public static function getType()
     {
         return 'snowflake';
     }
+
+    public function getCredentials()
+    {
+        $credentials = parent::getCredentials();
+        $credentials['account'] = $this->parseAccount($credentials['host']);
+
+        return $credentials;
+    }
+
+    /**
+     * Parses `account` from `host`
+     *
+     * Based on how Snowflake Python Connector handles account resolution.
+     * https://github.com/snowflakedb/snowflake-connector-python/blob/main/src/snowflake/connector/util_text.py#L242
+     */
+    private function parseAccount(string $host): string
+    {
+        $hostParts = explode('.', $host);
+
+        if (count($hostParts) <= 1) {
+            return $host;
+        }
+
+        if ($hostParts[1] !== 'global') {
+            return $hostParts[0];
+        }
+
+        return substr($hostParts[0], 0, strrpos($hostParts[0], '-'));
```

Comment:
can we test these conditions too?  maybe something that has `global` in the 1 slot and something that has a `-` in the 0 slot?

Reply:
I was thinking about that too, but came to conclusion that fully synthetic test, based just on the current implementation is not much of use - it does not tell us it's correct (correctly copied form the Snowflake Connector), just that the implementation hasn't changed over time. Ideally I would like to have a real life example of such host, but haven't find any so far.